src/mjolnir/restrictionbuilder.cc

```diff
@@ -239,17 +242,101 @@ std::deque<GraphId> GetGraphIds(GraphId& n_graphId,
   return graphids;
 }
 
+ComplexRestrictionBuilder CreateComplexRestriction(const OSMRestriction& restriction,
+                                                   const GraphId& from,
+                                                   const GraphId& to,
+                                                   const std::vector<GraphId>& vias) {
+  ComplexRestrictionBuilder complex_restriction;
+  complex_restriction.set_from_id(from);
+  complex_restriction.set_via_list(vias);
+  complex_restriction.set_to_id(to);
+  complex_restriction.set_type(restriction.type());
+  complex_restriction.set_modes(restriction.modes());
+
+  TimeDomain td = TimeDomain(restriction.time_domain());
+  if (td.td_value()) {
+    complex_restriction.set_begin_day_dow(td.begin_day_dow());
+    complex_restriction.set_begin_hrs(td.begin_hrs());
+    complex_restriction.set_begin_mins(td.begin_mins());
+    complex_restriction.set_begin_month(td.begin_month());
+    complex_restriction.set_begin_week(td.begin_week());
+    complex_restriction.set_dow(td.dow());
+    complex_restriction.set_dt(true);
+    complex_restriction.set_dt_type(td.type());
+    complex_restriction.set_end_day_dow(td.end_day_dow());
+    complex_restriction.set_end_hrs(td.end_hrs());
+    complex_restriction.set_end_mins(td.end_mins());
+    complex_restriction.set_end_month(td.end_month());
+    complex_restriction.set_end_week(td.end_week());
+  }
+
+  return complex_restriction;
+};
+
+struct Result {
+  uint32_t forward_restrictions_count;
+  uint32_t reverse_restrictions_count;
+  std::vector<ComplexRestrictionBuilder> restrictions;
+  std::unordered_set<GraphId> part_of_restriction;
+};
+
+void HandleOnlyRestrictionProperties(const std::vector<Result>& results,
+                                     const boost::property_tree::ptree& config) {
+  std::unordered_map<GraphId, std::vector<const ComplexRestrictionBuilder*>> restrictions;
+  std::unordered_map<GraphId, std::vector<GraphId>> part_of_restriction;
+  for (const auto& res : results) {
+    for (const auto& restriction : res.restrictions) {
+      restrictions[restriction.to_graphid().Tile_Base()].push_back(&restriction);
+    }
+    for (const auto& edge_id : res.part_of_restriction) {
+      part_of_restriction[edge_id.Tile_Base()].push_back(edge_id);
+    }
+  }
+
+  GraphReader reader(config);
+  for (const auto& i : restrictions) {
+    GraphId tile_id = i.first;
+    auto tile = reader.GetGraphTile(tile_id);
+    if (!tile)
+      continue;
+
+    GraphTileBuilder tile_builder(reader.tile_dir(), tile_id, true);
+    for (auto restriction : i.second) {
+      tile_builder.AddForwardComplexRestriction(*restriction);
```

Comment:
Add new prohibitory restriction that we couldn't handle within the main `build` function